### PR TITLE
Add some basic docs for data loading.

### LINKED ISSUE #21: Document current and intended data flow
We are currently loading historical data from the PeMS clearing house (cf #20, #18), but it isn't very well documented yet.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Fixes #21 

This is pretty terse, but gets us started on some docs for how data is flowing (or is intended to flow) for this project. @pingpingxiu-DOT-ca-gov and @kengodleskidot, I'd appreciate your eyes on this.

The diagram gets rendered like this:

```mermaid
flowchart TB
CD{Caltrans Districts}
subgraph Caltrans Network
  LCS((Land Closure\nSystem))
  CHP((CHP\nIncidents))
  L((Data Landing\nServer))
  DR((Data Relay\nServer))
  C((PeMS\nWebserver))
  OD[(Oracle data\nlake)]
  OW[(Oracle data\nwarehouse)]
end
subgraph ODI AWS Account
    MWAA((AWS Hosted\nAirflow))
    S3[S3 Bucket]
end
subgraph Snowflake
    RAW[(RAW)]
    TRANSFORM[(TRANSFORM)]
    ANALYTICS[(ANALYTICS)]
end
NC((New PeMS\nWebserver))
OP{Open Data}
RESEARCH{Researchers}
STAFF{Caltrans Traffic\nOperations}
CD --> L
C -. Historical\nData .- MWAA -.-> S3
L -. Iteris\nScripts .-> OD
OD -. Iteris\nScripts .-> OW
S3 -- Snowpipe --> RAW
OW -.-> C
OD -.-> C
L --> DR
DR --> S3
RAW --> TRANSFORM --> ANALYTICS
LCS --> DR
CHP --> DR
DR --> RAW
OD -.-> DR
OW -.-> DR
ANALYTICS --> OP
ANALYTICS --> NC
NC --> RESEARCH
NC --> STAFF
```

### CODE REVIEW

**Comment by pingpingxiu-DOT-ca-gov** [comment]
Great presentation @ian-r-rose ! I would love to learn the software tool to make the diagram!

And the overall graph grasped the whole picture nicely!

A couple comments: 

* From the Data Relay server out, there are two arrows. I think they would first going through some transformation process before landing in the S3 or snowflake. 
* Analytics may be first landing in the Web Service, and the downstream will interact with Web Service rather than directly on the Analytics. Snowflake may expose API to the Web Service using the service account, and the Web Service will manage the user database to grant access to end users on their authorized data reports. 

cc: @kengodleskidot @ZhenyuZhu-Caltrans

**Comment by ian-r-rose** [comment]
Thanks for the quick review @pingpingxiu-DOT-ca-gov!

> Great presentation @ian-r-rose ! I would love to learn the software tool to make the diagram!

This is created using [mermaidjs](https://mermaid.js.org/), which integrates nicely with GitHub and markdown.

> A couple comments:
> 
>     * From the Data Relay server out, there are two arrows. I think they would first going through some transformation process before landing in the S3 or snowflake.

My hope is still that we can do some light transformation and batching using the data relay server, as part of that DAG. We can prototype some things using our MWAA instance, but I still think ultimately it's a more natural fit in the data relay server (e.g., we can trigger that transformation after the tar file is uploaded, whereas there isn't a natural way to trigger the DAG on the MWAA side). But let's discuss further!

> 
>     * Analytics may be first landing in the Web Service, and the downstream will interact with Web Service rather than directly on the Analytics. Snowflake may expose API to the Web Service using the service account, and the Web Service will manage the user database to grant access to end users on their authorized data reports.

Good point, I've added a new webserver to the diagram, what do you think of that?

**Comment by pingpingxiu-DOT-ca-gov** [comment]
Don't know how to quote yet. 
Thanks for you intro on mermaidjs! I'll learn to use it. 

Regarding airflow triggering the transformation, I agree, I can add one task right after the uploading task to remote trigger your process to transform the data. 

Yes, the webserver modification on the graph looks good to me. 



@ian-r-rose

File: docs/data-loading.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ANALYTICS --> OP
+ANALYTICS --> NC
+NC --> RESEARCH
+NC --> STAFF
+```
+
+We are shifting on-premise data pipelines to the cloud,
+and getting data transformation logic into version control.
+
+The above diagram shows the both the prior state and the new architecture,
```
Comment (britt-allen) [inline docs/data-loading.md:74]:
```suggestion
The above diagram shows both the prior state and the new architecture,
```
deleted the second instance of "the"

**Review by britt-allen — APPROVED** [review]
This was a helpful write up of our data loading and architecture current state and goals. Thank you for putting this together, Ian!

**Pull request merged**